testing/tests/cpp_nesting.py

```diff
@@ -26,114 +41,128 @@ class CPPNesting(ExhaleTestCase):
-                clike("struct", "params"): {
-                    union("four_bytes"): {}
-                }
-            },
-            union("four_bytes"): {},
-            namespace("dual_nested"): {
-                clike("struct", "one"): {
-                    clike("struct", "params"): {
-                        union("four_bytes"): {}
-                    }
-                },
-                clike("struct", "two"): {
-                    clike("struct", "params"): {
-                        union("four_bytes"): {}
-                    }
-                }
-            }
-        }
-    }
-
     def test_hierarchies(self):
         """Verify the class and file hierarchies."""
-        compare_file_hierarchy(self, file_hierarchy(self.file_hierarchy_dict))
-        compare_class_hierarchy(self, class_hierarchy(self.class_hierarchy_dict))
+        compare_class_hierarchy(self, class_hierarchy(self.class_hierarchy_dict()))
+        compare_file_hierarchy(self, file_hierarchy(self.file_hierarchy_dict()))
 
     @confoverrides(exhale_args={"doxygenStripFromPath": "../include"})
     def test_hierarchies_stripped(self):
         """
         Verify the class and file hierarchies with ``doxygenStripFromPath=../include``.
-
-        .. todo:: this test is not supported yet
         """
-        return  # TODO: Exhale should remove the include/ directory
+        compare_class_hierarchy(self, class_hierarchy(self.class_hierarchy_dict()))
         # dirty hack to pop off the first include/ directory without needing to know
         # the actual object that is the first and only key
-        for key in self.file_hierarchy_dict:
-            no_include = self.file_hierarchy_dict[key]
+        file_hierarchy_dict = self.file_hierarchy_dict()
+        for key in file_hierarchy_dict:
+            no_include = file_hierarchy_dict[key]
             break
         compare_file_hierarchy(self, file_hierarchy(no_include))
-        compare_class_hierarchy(self, class_hierarchy(self.class_hierarchy_dict))
+
+
+def with_page_hierarchy(f_path, page_hierarchy):
+    """
+    Return a function using a generated fixture creating a page hierarchy.
+
+    The generated fixture generates a fixture that will create ``f_path`` prior to the
+    test running (so Doxygen sees it) and then afterward deletes the file.  The comment
+    generated will follow the provided ``page_hierarchy``.
+
+    **Parameters**
+    ``f_path`` (:class:`python:str`)
+        The destination to generate the C++ documented file encoding the specified
+        ``page_hierarchy``.
+
+    ``page_hierarchy`` (:class:`python:dict`)
+        A dictionary of :ref:`~testing.hierarchies.mainpage`,
+        :ref:`~testing.hierarchies.page`, and/or :ref:`~testing.hierarchies.subpage`
+        keys that either map to a :ref:`~testing.hierarchies.page_contents` node, or
+        a subpage.
+
+    **Return**
+    The input ``func``, after executing ``pytest.mark.usefixtures``.
```

Comment:
This docstr is throwing "unexpected indent" errors in sphinx. It should use sphinx compatible rst syntax.
```suggestion
    :param str f_path: The destination to generate the C++ documented file encoding the
        specified ``page_hierarchy``.
    :param dict page_hierarchy: A dictionary of :ref:`~testing.hierarchies.mainpage`,
        :ref:`~testing.hierarchies.page`, and/or :ref:`~testing.hierarchies.subpage`
        keys that either map to a :ref:`~testing.hierarchies.page_contents` node, or
        a subpage.

    :Returns: The input ``func``, after executing ``pytest.mark.usefixtures``.
```